testcases/kernel/syscalls/mincore/mincore02.c

```diff
@@ -1,136 +1,74 @@
- * along with this program;  if not, write to the Free Software
- * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ * Copyright (C) 2024 SUSE LLC Andrea Manzini <andrea.manzini@suse.com>
  */
 
-/*
+/*\
+ * [Description]
  * This test case provides a functional validation for mincore system call.
  * We mmap a file of known size (multiple of page size) and lock it in
  * memory. Then we obtain page location information via mincore and compare
  * the result with the expected value.
  */
 
-#include <sys/mman.h>
-#include <sys/types.h>
-#include <sys/stat.h>
-#include <sys/wait.h>
-#include <fcntl.h>
-#include <unistd.h>
-#include <signal.h>
-#include <errno.h>
+#include "tst_test.h"
 
-#include "test.h"
-#include "safe_macros.h"
+#define NUM_PAGES 4
 
-char *TCID = "mincore02";
-int TST_TOTAL = 1;
-
-static int fd = 0;
-static void *addr = NULL;
+static void *addr;
+static int size;
 static int page_size;
-static int num_pages = 4;
-static unsigned char *vec = NULL;
+
+static unsigned char vec[NUM_PAGES];
+static int fd;
 
 static void cleanup(void)
 {
-	free(vec);
-	munlock(addr, page_size * num_pages);
-	munmap(addr, page_size * num_pages);
-	close(fd);
-	tst_rmdir();
+	SAFE_MUNLOCK(addr, size);
+	SAFE_MUNMAP(addr, size);
+	if (fd)
+		close(fd);
 }
 
 static void setup(void)
 {
-	char *buf;
-	size_t size;
-
-	tst_tmpdir();
-
 	page_size = getpagesize();
-	if (page_size == -1)
-		tst_brkm(TBROK | TERRNO, cleanup, "Unable to get page size");
-
-	size = page_size * num_pages;
-	buf = malloc(size);
+	size = page_size * NUM_PAGES;
+	char buf[size];
 
 	memset(buf, 42, size);
-	vec = malloc((size + page_size - 1) / page_size);
-	
-	fd = SAFE_OPEN(cleanup, "mincore02", O_CREAT | O_RDWR,
-		       S_IRUSR | S_IWUSR);
-
-	/* fill the temporary file with two pages of data */
-	if (write(fd, buf, size) < 0) {
-		tst_brkm(TBROK | TERRNO, cleanup,
-		         "Error in writing to the file");
-	}
-	free(buf);
-
-	addr = mmap(NULL, size, PROT_READ | PROT_WRITE | PROT_EXEC,
-	            MAP_SHARED, fd, 0);
-
-	if (addr == MAP_FAILED) {
-		tst_brkm(TBROK | TERRNO, cleanup,
-                         "Unable to map file for read/write");
-	}
+
+	int fd = SAFE_OPEN("mincore02", O_CREAT | O_RDWR, 0600);
+
+	/* fill the temporary file with $NUM_PAGES pages of data */
+	SAFE_WRITE(SAFE_WRITE_ALL, fd, buf, size);
+	addr = SAFE_MMAP(NULL, size, PROT_READ | PROT_WRITE | PROT_EXEC, MAP_SHARED, fd, 0);
 
 	/* lock mmapped file, so mincore returns "in core" for all pages */
-	if (mlock(addr, size) == -1)
-		tst_brkm(TBROK | TERRNO, cleanup, "Unable to lock the file");
+	SAFE_MLOCK(addr, size);
 }
 
-int main(int argc, char **argv)
+static void check_mincore(void)
 {
-	int lock_pages, counter;
-	int lc;
-
-	tst_parse_opts(argc, argv, NULL, NULL);
-
-	setup();
-
-	for (lc = 0; TEST_LOOPING(lc); lc++) {
-		tst_count = 0;
+	TST_EXP_PASS_SILENT(mincore(addr, NUM_PAGES * page_size, vec));
```

Comment:
We are testing `mincore`, so rather than silent pass, we need a pass check as well